File: lib/172.factorial_trailing_zeroes.dart
void main() {
  print(trailingZeroes(3));
  print(trailingZeroes(5));
  print(trailingZeroes(0));
  print(trailingZeroes(10));
  print(trailingZeroes(11));
  print(trailingZeroes(12));
  print(trailingZeroes(13));
  print(trailingZeroes(14));
  print(trailingZeroes(15));
  print(trailingZeroes(25));
}

/// todo
int trailingZeroes(int n) {
  var result = 0;
  var currentFactor = 5;

  while (currentFactor <= n) {
    result += n ~/ currentFactor;
    currentFactor *= 5;
  }

  return result;
}

File: src/172.factorial_trailing_zeroes.kt
fun main() {
    println(trailingZeroes(3))
    println(trailingZeroes(5))
    println(trailingZeroes(0))
    println(trailingZeroes(10))
    println(trailingZeroes(11))
    println(trailingZeroes(12))
    println(trailingZeroes(13))
    println(trailingZeroes(14))
    println(trailingZeroes(15))
}

/// todo
fun trailingZeroes(n: Int): Int {
    var result: Int = 0
    var currentFactor: Int = 5

    while (currentFactor <= n) {
        result += n / currentFactor
        currentFactor *= 5
    }

    return result
}
